flag-app/flag-app-project/src/components/CardDetail.tsx

```diff
@@ -0,0 +1,75 @@
+import { FaArrowLeft } from "react-icons/fa6";
+import { useDarkMode } from "../hooks";
+import { BorderCountry } from "./BorderCountry";
+import CardItem from "./CardItem";
+import { Country } from "../interfaces";
+
+interface Props {
+  handleClickBack: () => void;
+  country: Country;
+}
+
+export const CardDetail = ({ handleClickBack, country }: Props) => {
+  const { mode } = useDarkMode();
+
+  const { name, population, region, capital, tld, currencies, languages } =
+    country;
+
+  const countryNameCode = Object.keys(name.nativeName)[0];
+  const nativeName = name.nativeName[countryNameCode].official;
+
+  const domain = tld[0];
+
+  const countryCurrencyCode = Object.keys(currencies)[0];
+  const currency = currencies[countryCurrencyCode].name;
+
+  const languageCode = Object.keys(languages)[0];
+  const language = languages[languageCode];
+
+  return (
+    <div>
+      <div
+        onClick={handleClickBack}
+        className={`inline-flex items-center gap-2 border rounded-md shadow-md mb-20 cursor-pointer px-6 py-1 ${
```

Comment:
This is kind of hard to read, so what I would suggest is:

1. Move all "fixed" class names into a single const: `const baseClassName = inline-flex items-center gap-2 border rounded-md shadow-md mb-20 cursor-pointer px-6 py-1`
2. Also move the mode: `const modeClassName = mode === 'dark' ? 'shadow-white' : ''`
3. Change this className to `{$baseClassName} {$modeClassName}`